examples/gdp/constrained_layout/cons_layout_model.py

```diff
@@ -148,3 +255,92 @@ def rectangle_in_circle(m, r):
         ]
 
     return m
+
+
+def draw_model(m, title=None):
+    """Draw a model using matplotlib to illustrate what's going on. Pass 'title' kwarg to give chart a title"""
+
+    # matplotlib setup
+    import matplotlib as mpl
+    import matplotlib.pyplot as plt
+
+    fig, ax = plt.subplots(1, 1, figsize=(10, 10))
+    ax.set_xlabel("x")
+    ax.set_ylabel("y")
+
+    # Hardcode these bounds since I'm not sure the best way to do it automatically
+    ax.set_xlim([0, 100])
+    ax.set_ylim([0, 100])
+
+    if title is not None:
+        plt.title(title)
+
+    for c in m.circles:
+        print(
+            f"drawing circle {c}: x={m.circle_x[c]}, y={m.circle_y[c]}, r={m.circle_r[c]}"
+        )
+        # no idea about colors
+        ax.add_patch(
+            mpl.patches.Circle(
+                (m.circle_x[c], m.circle_y[c]),
+                radius=m.circle_r[c],
+                facecolor="#0d98e6",
+                edgecolor="#000000",
+            )
+        )
+        ax.text(
+            m.circle_x[c],
+            m.circle_y[c] + m.circle_r[c] + 1.5,
+            f"Circle {c}",
+            horizontalalignment="center",
+        )
+
+    for r in m.rectangles:
+        print(
+            f"drawing rectangle {r}: x={m.rect_x[r]()}, y={m.rect_y[r]()} (center), L={m.rect_length[r]}, H={m.rect_height[r]}"
+        )
+        ax.add_patch(
+            mpl.patches.Rectangle(
+                (
+                    m.rect_x[r]() - m.rect_length[r] / 2,
+                    m.rect_y[r]() - m.rect_height[r] / 2,
+                ),
+                m.rect_length[r],
+                m.rect_height[r],
+                facecolor="#fbec68",
+                edgecolor="#000000",
+            )
+        )
+        ax.text(
+            m.rect_x[r](),
+            m.rect_y[r](),
+            f"R{r}",
+            horizontalalignment="center",
+            verticalalignment="center",
+            fontsize=8,
+        )
+
+    plt.show()
+
+
+# Solve some example problems and draw some pictures
+if __name__ == "__main__":
+    from pyomo.environ import SolverFactory
+    from pyomo.core.base import TransformationFactory
+
+    # Set up a solver, for example scip and bigm works
+    solver = SolverFactory("scip")
+    transformer = TransformationFactory("gdp.bigm")
+
+    # Do all of them
+    for d in ["l1", "l2"]:
+        for key in constrained_layout_model_examples.keys():
+            print(f"Solving example problem: {key}")
+            model = build_constrained_layout_model(
+                constrained_layout_model_examples[key], metric=d
+            )
+            transformer.apply_to(model)
+            solver.solve(model)
+            print(f"Found objective function value: {model.min_dist_cost()}")
+            draw_model(model, title=(key if d == "l1" else f"{key} ({d} distance)"))
```

Comment:
Minor nit, but why not put which norm in the title all the time?

Reply:
Hm. I think I wrote it like this because the l1 norm was the "default" problem specified in the paper and the euclidean distance was a variant invented by me. I can change it though, it should only make things clearer.